tests/integrations/config.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, List, Optional
+
+from pydantic import BaseModel
+
+
+__all__ = ["Config"]
+
+
+class Config:
+    """
+    Class containing all the configurations for a single stage of an integration
+
+    :param args_class: pydantic class for the CLI args for this stage
+    :param config: the raw config dictionary read in with yaml
+    :param command_stub: the base CLI command used to run the stage. A value of None
+        indicates that this stage doesn't have a subprocess script to be run
+    """
+
+    def __init__(
+        self, args_class: BaseModel, config: Dict, command_stub: Optional[str]
+    ):
+        self.raw_config = config
+        # pre_args are CLI commands that can precede the python call
+        # e.g. "torch.distributed.launch" or "CUDA_VISIBLE_DEVICES": 2
+        self.pre_args = config.pop("pre_args", "")
+        self._args_class = args_class
+        self.run_args = args_class(**config.get("command_args"))
+        self.command_stub = command_stub
+        # test args are used to guide testing of the stage. e.g. target metrics or
+        # named quantities/qualities to test for
+        self.test_args = config.pop("test_args", {})
+        # whether to replace '_' with '-' for run keywords
+        self.dashed_keywords = False
+        self._validate_config()
+
+    def create_command_script(self):
+        """
+        Handles logic for converting pydantic classes into valid argument strings.
+        This should set arg standards for all integrations and should generally not
+        be overridden. If the need to override comes up, consider updating this method
+        instead.
+
+        :return: string of the full CLI command
+        """
+        args_dict = self.run_args.dict()
+        args_string_list = []
+        for key, value in args_dict.items():
+            key = "--" + key
+            key = key.replace("_", "-") if self.dashed_keywords else key
+            # Handles bool type args (e.g. --do-train)
+            if isinstance(value, bool):
+                if value:
+                    args_string_list.append(key)
+            elif isinstance(value, List):
+                # Handles args that are both bool and value based (see evolve in yolov5)
+                if isinstance(value[0], bool):
```

Comment:
Potential Error if value is []

```suggestion
                if value and isinstance(value[0], bool):
```